shuup/front/static_src/js/pagination.js

```diff
@@ -22,11 +26,36 @@ window.moveToPage = function moveToPage(pageNumber) {
         }
     }
     window.PAGE_NUMBER = pageNumber;
+    reloadProducts(filterString);
 
-    reloadProducts();
+    if(window.history && window.history.pushState) {
+        history.pushState(state, null, filterString);
+    }
+    // prevent scroll to page buttons
+    return false;
 };
 
-function reloadProducts() {
-    var filterString = "?sort=" + $("#id_sort").val() + "&page=" + window.PAGE_NUMBER;
+function getFilterString(state) {
+    var filterString = "";
+    if(state !== null) {
+        filterString = "?sort=" + state.sort + "&page=" + state.page;
+    }
+    return filterString;
+}
+
+function reloadProducts(filterString) {
+    // this is to ensure browser back/forward from different domain does a full refresh
+    if(filterString == "") {
+        filterString += "?ajax=1";
+    } else {
+        filterString += "&ajax=1";
+    }
```

Comment:
to dry this out a bit I'd do 
``` javascript
filterString += "?" ? (filterString == "") : "&";
filterString += "ajax=1";
```
but it might lose readability a bit :)